components/eamxx/src/control/atmosphere_driver.cpp

```diff
@@ -128,7 +128,21 @@ set_params(const ekat::ParameterList& atm_params)
   fvphyshack = m_atm_params.sublist("grids_manager").get<std::string>("physics_grid_type", "None") == pg_type;
   if (fvphyshack) {
     // See the [rrtmgp active gases] note in share/util/eamxx_fv_phys_rrtmgp_active_gases_workaround.hpp
-    fv_phys_rrtmgp_active_gases_init(m_atm_params);
+    // To figure out whether we're using RRTMGP, we create a process group and
+    // see whether RRTMGP is in it. This seems heavy-handed, but it reduceѕ code
+    // duplication.
+    bool using_rrtmgp = false;
+    auto& atm_proc_params = m_atm_params.sublist("atmosphere_processes");
+    auto group = std::make_shared<AtmosphereProcessGroup>(m_atm_comm,atm_proc_params);
+    for (int i = 0; i < group->get_num_processes(); ++i) {
+      if (group->get_process(i)->name() == "rrtmgp") {
```

Comment:
This assumes that there are no nested groups. In practice, we _do_ have nested groups. In particular, rrtmgp is part of the "physics" atm proc group, so this check will not detect rrtmgp.

I think the best way to deal with this through recursion. You can either wrap this check in a lambda, or add a method to the atm proc group class.